tests/test_discriminated_union.py

```diff
@@ -1696,3 +1698,68 @@ class DiscriminatedModel(BaseModel):
             ]
     except PydanticUserError as exc_info:
         assert exc_info.code == 'callable-discriminator-no-tag'
+
+
+def test_presence_of_discriminator_when_generating_type_adaptor_json_schema_definitions() -> None:
+    class ItemType(str, Enum):
+        ITEM1 = 'item1'
+        ITEM2 = 'item2'
+
+    class CreateItem1(BaseModel):
+        item_type: Annotated[Literal[ItemType.ITEM1], Field(alias='type')]
+        id: int
+
+    class CreateItem2(BaseModel):
+        item_type: Annotated[Literal[ItemType.ITEM2], Field(alias='type')]
+        id: int
+
+    class CreateObjectDto(BaseModel):
+        id: int
+        items: List[
+            Annotated[
+                Union[
+                    CreateItem1,
+                    CreateItem2,
+                ],
+                Field(discriminator='item_type'),
+            ]
+        ]
+
+    adaptor = TypeAdapter(
+        Annotated[CreateObjectDto, FieldInfo(examples=[{'id': 1, 'items': [{'id': 3, 'type': 'ITEM1'}]}])]
```

Comment:
Why use FieldInfo here instead of just Field()?

Reply:
The example provided to explain the issue used `Body` class in fastapi which inherits from the `FieldInfo` class in pydantic. As `FieldInfo` was the closest classes I used it.

Reply:
Gotcha okay!